gzsitl/gzsitl_plugin.cc

```diff
@@ -102,22 +189,18 @@ void GZSitlPlugin::OnUpdate()
             return;
         }
 
-        if (!is_guided) {
-            mavserver.set_mode_guided();
+        if (mod != mode::GUIDED) {
+            this->mav->set_mode(mode::GUIDED);
             return;
         }
 
-        if (!is_armed) {
```

Comment:
I'm not sure how I would use a switch here. I need to check sequentially a series of flags to see if a specific command has to be sent/resent.